sql/hive/src/main/scala/org/apache/spark/sql/hive/client/package.scala

```diff
@@ -103,6 +103,7 @@ package object client {
     // Since HIVE-14496, Hive materialized view need calcite-core.
     // For spark, only VersionsSuite currently creates a hive materialized view for testing.
     case object v2_3 extends HiveVersion("2.3.7",
+      extraDeps = Seq("org.apache.zookeeper:zookeeper:3.4.6"),
```

Comment:
Although we have this, the GitHub Action is stilling failing. Maybe, we have other issues?

Reply:
> Although we have this, the GitHub Action is stilling failing. Maybe, we have other issues?

Yes,  with zookeeper dependency but ivy download it with it's test jars, after detail check I can't find code issue related to this problem. so strange.

Reply:
> Although we have this, the GitHub Action is stilling failing. Maybe, we have other issues?

Since default hive version is 2.3.7 and 2.3.7 can pass jdk9+, so. I changed hive version. in HiveExternalCatalogVersionSuite.   It's reasonable.

So confused, still working to find why it download zookeeper(test-jar).

Reply:
In maven repo I noticed that only zokeeper-3.4.x have <type>pom</type>,  but  it doesn't seem like this is the problem
![image](https://user-images.githubusercontent.com/46485123/99882466-a661b780-2c5b-11eb-893f-33d785d35d14.png)

Reply:
I agree that it's weird. Maybe, could you try this `Spark Shell`? If we starts without local caches (Maven/Ivy), I guess `Spark Shell` will fail with the same reason.

Reply:
Then, we should fix it first as an independent JIRA issue.

Reply:
I tried in feature https://github.com/apache/spark/pull/29966 to download zookeeper-3.4.6(after clean cache)
```
test("SPARK-33084: Add jar support ivy url") {
    sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local-cluster[3, 1, 1024]"))
    sc.addJar("ivy://org.apache.zookeeper:zookeeper:3.4.6")
    println("+===================+")
    sc.listJars().foreach(println)
    assert(sc.listJars().find(_.contains("zookeeper-3.4.6")).nonEmpty)
  }
```
It really download right jar.
But with UT
```
 test("SPARK-33084: Add jar support ivy url") {
    sc = new SparkContext(new SparkConf().setAppName("test")
      .setMaster("local-cluster[3, 1, 1024]"))
    sc.addJar("ivy://org.apache.hive:hive-exec:2.3.7" +
      "?exclude=org.pentaho:pentaho-aggdesigner-algorithm&transitive=true")
    println("+===================+")
    sc.listJars().foreach(println)
  }
```
It download both zookeeper.jar and zookeeper-test.jar.

Reply:
@dongjoon-hyun 
This error happened when  `HiveExternalCatalogVersionsSuite`.`prepare()`, So any change in current code won't work since it use downloaded compiled spark.
So change in current code won't change the result.